documentation/modules/cruise-control/con-rebalance-goals-proposals.adoc

```diff
@@ -0,0 +1,180 @@
+Ignore this goal if doing so will enable all hard goals to be met.
+
+¦Hard goals
+
+¦Hard goals _must_ be satisfied in rebalance proposals. 
+In other words, a rebalance proposal that does _not_ satisfy all the designated hard goals is rejected by Cruise Control and not sent to the user.
+
+Designating hard goals is optional.
+¦All brokers in the cluster must have fewer than 3 replicas (replica capacity goal).
+
+|===
+
+[discrete]
+=== Default and custom rebalance goals
+
+When interacting with Cruise Control to generate rebalance proposals, you can use the _default rebalance goals_, which are defined on deployment, or set ad hoc _custom rebalance goals_. 
+Both types of rebalance goal can be configured to include hard goals as well as soft goals.  
+
+*Default rebalance goals and cached proposal*
+
+The default rebalance goals are set in the `Kafka` custom resource. 
+You specify a list of goals in `spec.cruiseControl.config.default.goals`.
+
+Unless you set custom rebalance goals, the default rebalance goals are used by Cruise Control to return a _cached rebalance proposal_. 
+The cached proposal reflects the current state of the Kafka cluster and is updated every 15 minutes by default. 
+
+To review the list of default goals, see link:https://github.com/linkedin/cruise-control/wiki/Configurations#analyzer-configurations[Analyzer Configurations^] in the Cruise Control Wiki.
+
+For more information on setting the default rebalance goals, see xref:ref-cruise-control-configuration-{context}[]. 
+
+*Custom rebalance goals* 
+
+Custom rebalance goals are set in the `spec.goals` section of a `KafkaClusterRebalance` resource and override the default rebalance goals. 
+If no custom rebalance goals are set, the default rebalance goals are used.
+
+Use custom rebalance goals to generate a rebalance proposal that addresses a particular optimization scenario. 
+For example, you might want to optimize leader replica distribution across the Kafka cluster without considering other goals, such as disk or CPU usage. 
+So, you create a `KafkaClusterRebalance` resource containing a custom rebalance goal for leader replica distribution only.
+
+Using custom rebalance goals often returns a rebalance proposal faster than the default rebalance goals. 
+They must always be a subset of the default rebalance goals.
+
+For more information on setting custom rebalance goals, see xref:proc-generating-rebalance-proposals-{context}[].
+
+[discrete]
+== Rebalance proposals
+
+A _rebalance proposal_ is a summary of proposed changes from Cruise Control that would produce a more balanced Kafka cluster, with partition workloads distributed more evenly across the brokers. 
+Each rebalance proposal is based on the set of rebalance goals (either default goals or custom goals) that were used to generate it.
+
+Rebalance proposals always satisfy all the supplied hard goals, though not necessarily all the supplied soft goals.
```

Comment:
Again you can set a skip hard goals check flag in the `KafkaRebalace` custom resource.